### PR TITLE
Ignore kservices not created by serverless-controller-manager in revision-controller

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- as in title, small optimization

**Related issue(s)**

See https://github.com/kyma-project/kyma/issues/8254

### CODE REVIEW

File: components/function-controller/internal/controllers/knative/service_controller_test.go [hunk 1]
```diff
@@ -146,10 +153,18 @@ var _ = ginkgo.Describe("KService controller", func() {
 		reconciler = NewServiceReconciler(resourceClient, log.Log, config)
 		request = ctrl.Request{NamespacedName: types.NamespacedName{Namespace: namespace, Name: srvName}}
 	})
+	ginkgo.AfterEach(func() {
+		// DeleteAllBySelector(ctx, &servingv1.Revision{}, service.GetNamespace(), selector)
```
Comment (aerfio) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:157]:
```suggestion
```

Comment (aerfio) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:157]:
```suggestion
```

Comment (aerfio) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:157]:
```suggestion
```

File: components/function-controller/internal/controllers/knative/service_controller.go [hunk 2]
```diff
@@ -68,6 +69,11 @@ func (r *ServiceReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !hasCorrectLabels(*instance) {
+		r.Log.WithValues("service", request.NamespacedName).Info("skipping reconcilation for a service without needed labels")
+		return ctrl.Result{}, nil
+	}
+
```
Comment (michal-hudy) [inline components/function-controller/internal/controllers/knative/service_controller.go:76]: It should be also added to predicate

File: components/function-controller/internal/controllers/knative/service_controller.go [hunk 3]
```diff
@@ -68,6 +69,11 @@ func (r *ServiceReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error)
...
 	if !instance.Status.IsReady() {
 		return ctrl.Result{RequeueAfter: r.config.RequeueDuration}, nil
 	}
```
Comment (michal-hudy) [inline components/function-controller/internal/controllers/knative/service_controller.go:79]: I know it is not related to this PR, but it doesn't make sense to add here `RequeueAfter` as on Service Instance status change controller will get update event.

`RequeueAfter` should be a default result from `Reconcile` method when all operations finished sucessufull, or no operation needed (revisions are cleaned)

Comment (aerfio) [reply components/function-controller/internal/controllers/knative/service_controller.go:79]: ok, I will delete it

now I'm wondering whether we need `requeueAfter` at all, as we operate on events

Comment (michal-hudy) [reply components/function-controller/internal/controllers/knative/service_controller.go:79]: As a default return from `Reconcile` method

File: components/function-controller/internal/controllers/knative/service_controller_test.go [hunk 4]
```diff
@@ -146,10 +153,17 @@ var _ = ginkgo.Describe("KService controller", func() {
 		reconciler = NewServiceReconciler(resourceClient, log.Log, config)
 		request = ctrl.Request{NamespacedName: types.NamespacedName{Namespace: namespace, Name: srvName}}
 	})
+	ginkgo.AfterEach(func() {
+		sel := labels.SelectorFromSet(map[string]string{testCleanupLabelKey: testCleanupLabelValue})
+		err := resourceClient.DeleteAllBySelector(ctx, &servingv1.Revision{}, namespace, sel)
```
Comment (magicmatatjahu) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:158]: Revision are not deleted automatically after removing Service?

File: components/function-controller/internal/controllers/knative/service_controller_test.go [hunk 5]
```diff
@@ -210,11 +261,79 @@ func fixRevisionList(parentSvcName, namespace string, num int) []servingv1.Revis
 	return revList
 }
 
-func fixKservice(name, namespace string) servingv1.Service {
+func fixKservice(name, namespace string, addLables bool) servingv1.Service {
```
Comment (magicmatatjahu) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:264]: `addLabels` :P

File: components/function-controller/internal/controllers/knative/service_controller_test.go [hunk 6]
```diff
@@ -210,11 +261,79 @@ func fixRevisionList(parentSvcName, namespace string, num int) []servingv1.Revis
...
+	svcLabels := map[string]string{
+		testCleanupLabelKey: testCleanupLabelValue,
+	}
+
+	if addLables {
+		svcLabels = map[string]string{
+			serverlessv1alpha1.FunctionNameLabel:      "whatever-name",
+			serverlessv1alpha1.FunctionUUIDLabel:      "whatever-uuid",
+			serverlessv1alpha1.FunctionManagedByLabel: "whatever-managed-by",
+			testCleanupLabelKey:                       testCleanupLabelValue,
```
Comment (magicmatatjahu) [inline components/function-controller/internal/controllers/knative/service_controller_test.go:274]: Maybe push new labels, not override whole map? I see that in above lines you create default map with only this label.

File: components/function-controller/internal/controllers/knative/predicate.go [hunk 7]
```diff
@@ -23,6 +23,11 @@ func (r *ServiceReconciler) getPredicates() predicate.Predicate {
 				return false
 			}
 
+			if !hasCorrectLabels(*newSrv) {
+				log.Info("Skipping reconciling because kservice doesn't have correct labels", "name", newSrv.GetName(), "namespace", newSrv.GetNamespace(), "apiVersion", newSrv.GroupVersionKind().GroupVersion(), "kind", newSrv.GroupVersionKind().Kind)
+				return false
+			}
+
```
Comment (michal-hudy) [inline components/function-controller/internal/controllers/knative/predicate.go:30]: What about `GenericFunc` ?

File: components/function-controller/internal/controllers/knative/service_controller_test.go (hunk 4 above)
Comment (aerfio) [reply components/function-controller/internal/controllers/knative/service_controller_test.go:158]: Not when using test apiserver, we do not have knative controller, do not have k8s controller which handles owner references etc

File: components/function-controller/README.md [hunk 8]
```diff
@@ -40,3 +40,4 @@ The Function Controller uses these environment variables:
...
 | **APP_FUNCTION_DOCKER_ADDRESS**                   | Internal address of the Docker Registry                                                                                                                                | `serverless-docker-registry.kyma-system.svc.cluster.local:5000`                                                                                   |
 | **APP_FUNCTION_DOCKER_EXTERNAL_ADDRESS**          | External address of the Docker Registry                                                                | `registry.kyma.local`                                                                                                                                      |
+| **APP_KSERVICE_REQUEUE_DURATION**                 | Period of time after which the Knative Service Controller retries cleanup of stale revisions                 | `10m`                                                                                                                                      |
```
Comment (bszwarc) [inline components/function-controller/README.md:43]:
```suggestion
| **APP_KSERVICE_REQUEUE_DURATION**                 | Period of time after which the Knative Service Controller retries to clean up stale revisions                 | `10m`                                                                                                                                      |
```

**Pull request merged**